spec/requests/post_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts', type: :request do
+  describe 'index' do
+    before(:example) do
+      user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.',
+                         posts_counter: 0)
+      get "/users/#{user.id}/posts"
+    end
+
+    it 'response status was correct' do
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'correct template was rendered' do
+      expect(response).to render_template(:index)
+    end
+
+    it 'the response body includes correct placeholder text' do
+      expect(response.body).to include('All posts ')
+    end
+  end
+
+  before(:example) do
+    user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.',
+                       posts_counter: 0)
+    post = Post.new(author: user, title: 'Hello', text: 'This is my first post', comments_counter: 2,
+                    likes_counter: 2)
+    get "/users/#{user.id}/posts/#{post.id}"
+  end
+
+  it 'response status was correct' do
+    expect(response).to have_http_status(:ok)
+  end
+
+  it 'correct template was rendered' do
+    expect(response).to render_template(:index)
```

Comment:
* As stated in [the requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/controller_specs_project.md#:~:text=Make%20sure%20that%20for,includes%20correct%20placeholder%20text.) Kindly write a test for each action in your controllers ( which are `index` and `show` ) but here all tests are written for the index action only. kindly add a test for show action to tell If a correct template was rendered.

Reply:
Totally forgot the make the changes from the index to show. when I run the test when it passes I thought It was working. fixed it now. @Davidosky007 thank you for pointing that out.